### PR TITLE
Commander: Add land detection configuration to preflight checks

### PR DESCRIPTION
**Pull request by LorenzMeier**
This should help to avoid having users which set up the land detector up for in-air land detection.

@PX4TestFlights Please test.

### CODE REVIEW

File: src/modules/commander/PreflightCheck.cpp [hunk 1]
```diff
@@ -208,6 +208,29 @@ static bool imuConsistencyCheck(orb_advert_t *mavlink_log_pub, bool report_statu
 	return success;
 }
 
+static bool landConsistencyCheck(orb_advert_t *mavlink_log_pub, bool report_status)
+{
+	bool success = true; // start with a pass and change to a fail if any test fails
+
+	// Check the relationship between land detection threshold and land speed
+	float land_speed, land_detect_speed;
```
Comment (dagar) [inline src/modules/commander/PreflightCheck.cpp]: If MC or VTOL

Comment (LorenzMeier) [reply src/modules/commander/PreflightCheck.cpp]: That's true, but if that param is messed up on a fixed wing it would be equally disconcerting.

**Comment by dagar** [comment]
I'm wondering if it would be cleaner architecturally for the land detector (aka vehicle model) to report it's status instead and have commander simply enforce the requirement.

Then we can avoid having to piece together the logic in a completely separate module.

**Comment by LorenzMeier** [comment]
Yes, we should rework this PR to fulfil that.

**Comment by MaEtUgR** [comment]
As we're aiming to fix cases like #8600 I want to mention that we also had these false positives for the first stage of landing detection resulting in very short thrust drops when the hover thrust was completely off (was too high) because the land detector uses the parameters to set it's thresholds. I agree that having the checks in the same place would make it easier for the future.

**Pull request closed without merging**